### PR TITLE
Add TLS feature for ModbusTcpClient

### PR DESCRIPTION
**Pull request by starnight**
Modbus.org released MODBUS/TCP Security Protocol Specification [1],
which focuses variant of the Mobdbus/TCP protocol utilizing Transport
Layer Security (TLS). This patch enables the Modbus TCP over TLS feature
for ModbusTcpClient with the Python builtin module ssl - TLS/SSL wrapper
for socket objects.

[1]: http://modbus.org/docs/MB-TCP-Security-v21_2018-07-24.pdf

### CODE REVIEW

**Comment by dhoomakethu** [comment]
@starnight Thanks for the PR, Could you also add a simple example in the examples/contrib directory to use this feature ?

**Comment by starnight** [comment]
Added the simple example as examples/contrib/modbus_tcptls_client.py

File: examples/contrib/modbus_tcptls_client.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+---------------------------------------------------------------------------
+
+This is a simple example of writing a modbus TCP over TLS client that uses
+Python builtin module ssl - TLS/SSL wrapper for socket objects for the TLS
+feature.
+"""
+# -------------------------------------------------------------------------- #
+# import neccessary libraries
+# -------------------------------------------------------------------------- #
+import ssl
```
[the remark this answers is not in the document]
Comment (starnight) [reply examples/contrib/modbus_tcptls_client.py]: The standard ssl module https://docs.python.org/3/library/ssl.html

[the remark this answers is not in the document]
Comment (dhoomakethu) [reply examples/contrib/modbus_tcptls_client.py]: Thanks for the confirmation, I will go through the spec and review .

**Comment by starnight** [comment]
After one night over, I think I should make it as new class called ModbusTcpTlsClient which inherits ModbusTcpClient.  And over write constructor, connect, _recv and __str__ fuctions.  This may make codes clean and follow object oriented rules.
If this is a better idea, I can send a new patch for it.

**Comment by dhoomakethu** [comment]
Sure, that sounds good and will be easier to review. Thanks.

**Comment by starnight** [comment]
I also updated `test/test_client_sync.py` for the new **ModbusTlsClient** tests.

**Comment by starnight** [comment]
Oops!  I found one mistake which is I make all MODBUS TCP/IP ADU over TLS.  According to [MODBUS/TCP Security Protocol Specification](http://modbus.org/docs/MB-TCP-Security-v21_2018-07-24.pdf) Figure 3 mbap PDU Encapsulated in TLS, it should only be PDU over TLS.

I should fix this first.

**Comment by tracernz** [comment]
That's an interesting design choice. I could see it if mbaps was still on port 502 and they wanted some mbap/mbaps auto-negotiation mechanism... but it's on port 802.

**Comment by starnight** [comment]
Fixed with new tls_framer and submit the new pull request #446.

**Pull request closed without merging**